### PR TITLE
GH-36199: [Python][CI][Spark] Update spark versions used on our nightly tests

### LINKED ISSUE #36199: [Python][CI] Update spark versions tested on our nightly tests
### Describe the enhancement requested

It does seem we are using some pretty old and deprecated spark versions on our CI nightly integration builds with spark.
As discussed on this PR comment discussion: https://github.com/apache/arrow/pull/36061#discussion_r1230266720

The following comments were added:

Do we still need Spark 3.1.2?

I think that Spark 3.1 reached EOL:

https://spark.apache.org/versioning-policy.html

>    Feature release branches will, generally, be maintained with bug fix releases for a period of 18 months.

And 3.1.1 was released on 2021-03-02:

https://spark.apache.org/news/index.html

>    [Spark 3.1.1 released](https://spark.apache.org/news/spark-3-1-1-released.html)
>    March 2, 2021

3.2 might reach EOL because 3.2.0 was released on 2021-10-13:

>    [Spark 3.2.0 released](https://spark.apache.org/news/spark-3-2-0-released.html)
>    October 13, 2021

### Component(s)

Continuous Integration, Python

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

We are currently testing with very old or deprecated spark versions.

### What changes are included in this PR?

Update the spark versions to be tested.

### Are these changes tested?

They will via archery

### Are there any user-facing changes?

No
* Closes: #36199

### CODE REVIEW

**Comment by raulcd** [comment]
This is currently blocked by: https://github.com/apache/arrow/issues/36332

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -1588,9 +1588,9 @@ tasks:
       image: conda-python-hdfs
 {% endfor %}
 
-{% for python_version, spark_version, test_pyarrow_only, numpy_version in [("3.8", "v3.1.2", "false", "latest"),
-                                                                           ("3.9", "v3.2.0", "false", "1.23"),
-                                                                           ("3.10", "master", "false", "latest")] %}
+{% for python_version, spark_version, test_pyarrow_only, numpy_version in [("3.8", "v3.4.1", "false", "latest"),
+                                                                           ("3.10", "v3.4.0", "false", "1.23"),
```
Comment (kou) [inline dev/tasks/tasks.yml:1592]: Should we use the same version (`v3.4.1` or `v3.4.0`) here?

**Review by kou — APPROVED** [review]
+1

File: dev/tasks/tasks.yml (hunk 1 above)
Comment (raulcd) [reply dev/tasks/tasks.yml:1592]: I thought on covering several versions but now that I look into it again, a patch difference will not make a huge difference so I am ok changing it to:
```suggestion
{% for python_version, spark_version, test_pyarrow_only, numpy_version in [("3.8", "v3.4.1", "false", "latest"),
                                                                           ("3.10", "v3.4.1", "false", "1.23"),
```

**Review by assignUser — APPROVED** [review]

**Pull request merged**